tests/integration/components/file-dropzone-test.js

```diff
@@ -7,135 +7,223 @@ import {
-
-    await dragEnter('.test-dropzone');
-
-    assert.verifySteps(['onDragEnter']);
-  });
-
-  test('onDragLeave is called when a file is dragged out', async function (assert) {
-    this.onDragLeave = () => assert.step('onDragLeave');
-
-    await render(hbs`
-      <FileDropzone
-        class="test-dropzone"
-        @name="test"
-        @onDragLeave={{this.onDragLeave}} />
-    `);
-
-    await dragEnter('.test-dropzone', new File([], 'dingus.txt'));
-    await dragLeave('.test-dropzone', new File([], 'dingus.txt'));
-
-    assert.verifySteps(['onDragLeave']);
+  module('new api', function (hooks) {
+    hooks.beforeEach(function () {
+      const fileQueueService = this.owner.lookup('service:file-queue');
+      this.queue = new Queue({ name: 'test', fileQueue: fileQueueService });
+    });
+
+    test('filesEnter', async function (assert) {
+      this.filesEnter = (files) =>
+        files.forEach((file) => assert.step(file.name));
+      await render(hbs`
+        <FileDropzone
+          class="test-dropzone"
+          @queue={{this.queue}}
+          @filesEnter={{this.filesEnter}}
+        />
+      `);
+
+      await dragEnter('.test-dropzone', new File([], 'dingus.txt'));
+
+      assert.verifySteps(['dingus.txt']);
+    });
+
+    test('filesLeave', async function (assert) {
+      this.filesLeave = (files) =>
+        files.forEach((file) => assert.step(file.name));
+      await render(hbs`
+        <FileDropzone
+          class="test-dropzone"
+          @queue={{this.queue}}
+          @filesLeave={{this.filesLeave}}
+        />
+      `);
+
+      await dragEnter('.test-dropzone', new File([], 'dingus.txt'));
+      await dragLeave('.test-dropzone', new File([], 'dingus.txt'));
+
+      assert.verifySteps(['dingus.txt']);
+    });
+
+    test('filter and filesDropped', async function (assert) {
+      this.filter = (file) => file.name.includes('.txt');
+      this.filesDropped = (files) =>
+        files.forEach((file) => assert.step(file.name));
+      await render(hbs`
+        <FileDropzone
+          class="test-dropzone"
+          @queue={{this.queue}}
+          @filter={{this.filter}}
+          @filesDropped={{this.filesDropped}}
+        />
+      `);
+
+      await dragAndDrop(
+        '.test-dropzone',
+        new File([], 'dingus.txt'),
+        new File([], 'dangus.wmv'),
+        new File([], 'dongus.txt')
+      );
+
+      assert.verifySteps(['dingus.txt', 'dongus.txt']);
+    });
   });
 
-  test('yielded properties', async function (assert) {
-    await render(hbs`
-      <FileDropzone @name="test" as |dropzone queue|>
-        <div class="supported">{{dropzone.supported}}</div>
-        <div class="active">{{dropzone.active}}</div>
-        <div class="queue-name">{{queue.name}}</div>
-      </FileDropzone>
-    `);
-
-    assert.dom('.supported').hasText('true');
-    assert.dom('.active').hasText('false');
-    assert.dom('.queue-name').hasText('test');
+  module('deprecated api', function () {
```

Comment:
oooooh, that's a nice one to move that into a `module` and to have the chance to delete it at once later on 👍